File: src/Application Specific/Classes/Entity Collections/Order/SiteOrder/SiteOrderSQL.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace SiteOrders

Public Class SiteOrderSQL
    Private _database As Sys.Database

    Public Sub New(ByVal database As Sys.Database)
        _database = database
    End Sub

#Region "Functions"

	
    Public Sub Delete(ByVal SiteOrderID As Integer)
    		_database.ClearParameter()
            _database.AddParameter("@SiteOrderID", SiteOrderID, True)
            _database.ExecuteSP_NO_Return("SiteOrder_Delete")
            End Sub

            Public Sub DeleteByOrder(ByVal OrderID As Integer)
                _database.ClearParameter()
                _database.AddParameter("@OrderID", OrderID, True)
                _database.ExecuteSP_NO_Return("SiteOrder_DeleteByOrder")
            End Sub

            Public Function [SiteOrder_Get](ByVal SiteOrderID As Integer) As SiteOrder
                _database.ClearParameter()
                _database.AddParameter("@SiteOrderID", SiteOrderID)

                'Get the datatable from the database store in dt
                Dim dt As DataTable = _database.ExecuteSP_DataTable("SiteOrder_Get")

                If Not dt Is Nothing Then
                    If dt.Rows.Count > 0 Then
                        Dim oSiteOrder As New SiteOrder
                        With oSiteOrder
                            .IgnoreExceptionsOnSetMethods = True
                            .SetSiteOrderID = dt.Rows(0).Item("SiteOrderID")
                            .SetSiteID = dt.Rows(0).Item("SiteID")
                            .SetOrderID = dt.Rows(0).Item("OrderID")
                        End With
                        oSiteOrder.Exists = True
                        Return oSiteOrder
                    Else
                        Return Nothing
                    End If
                Else
                    Return Nothing
                End If
            End Function

            Public Function [SiteOrder_GetForOrder](ByVal OrderID As Integer) As SiteOrder
                _database.ClearParameter()
                _database.AddParameter("@OrderID", OrderID)

                'Get the datatable from the database store in dt
                Dim dt As DataTable = _database.ExecuteSP_DataTable("SiteOrder_GetForOrder")

                If Not dt Is Nothing Then
                    If dt.Rows.Count > 0 Then
                        Dim oSiteOrder As New SiteOrder
                        With oSiteOrder
                            .IgnoreExceptionsOnSetMethods = True
                            .SetSiteOrderID = dt.Rows(0).Item("SiteOrderID")
                            .SetSiteID = dt.Rows(0).Item("SiteID")
                            .SetOrderID = dt.Rows(0).Item("OrderID")
                        End With
                        oSiteOrder.Exists = True
                        Return oSiteOrder
                    Else
                        Return Nothing
                    End If
                Else
                    Return Nothing
                End If
            End Function

            Public Function [SiteOrder_GetAll]() As DataView
                _database.ClearParameter()
                Dim dt As DataTable = _database.ExecuteSP_DataTable("SiteOrder_GetAll")
                dt.TableName = Entity.Sys.Enums.TableNames.tblSiteOrder.ToString
                Return New DataView(dt)
            End Function

            Public Function [Insert](ByVal oSiteOrder As SiteOrder, ByVal trans As SqlClient.SqlTransaction) As SiteOrder

                Dim Command As SqlClient.SqlCommand = New SqlClient.SqlCommand

                Command.CommandText = "SiteOrder_Insert"
                Command.CommandType = CommandType.StoredProcedure
                Command.Transaction = trans
                Command.Connection = trans.Connection

                Command.Parameters.AddWithValue("@OrderID", oSiteOrder.OrderID)
                Command.Parameters.AddWithValue("@SiteID", oSiteOrder.SiteID)

                oSiteOrder.SetSiteOrderID = Entity.Sys.Helper.MakeIntegerValid(Command.ExecuteScalar)
                oSiteOrder.Exists = True
                Return oSiteOrder

            End Function

            Public Function [Insert](ByVal oSiteOrder As SiteOrder) As SiteOrder
                _database.ClearParameter()
                AddSiteOrderParametersToCommand(oSiteOrder)

                oSiteOrder.SetSiteOrderID = Entity.Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("SiteOrder_Insert"))
                oSiteOrder.Exists = True
                Return oSiteOrder
            End Function

            Public Sub [Update](ByVal oSiteOrder As SiteOrder)
                _database.ClearParameter()
                _database.AddParameter("@SiteOrderID", oSiteOrder.SiteOrderID, True)
                AddSiteOrderParametersToCommand(oSiteOrder)
                _database.ExecuteSP_NO_Return("SiteOrder_Update")
            End Sub



            Private Sub AddSiteOrderParametersToCommand(ByRef oSiteOrder As SiteOrder)
                With _database
                    .AddParameter("@SiteID", oSiteOrder.SiteID, True)
                    .AddParameter("@OrderID", oSiteOrder.OrderID, True)



                End With
            End Sub


#End Region

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Order/SiteOrder/SiteOrderSQL.cs
using System.Data;
using System.Data.SqlClient;

namespace FSM.Entity
{
    namespace SiteOrders
    {
        public class SiteOrderSQL
        {
            private Sys.Database _database;

            public SiteOrderSQL(Sys.Database database)
            {
                _database = database;
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            public void Delete(int SiteOrderID)
            {
                _database.ClearParameter();
                _database.AddParameter("@SiteOrderID", SiteOrderID, true);
                _database.ExecuteSP_NO_Return("SiteOrder_Delete");
            }

            public void DeleteByOrder(int OrderID)
            {
                _database.ClearParameter();
                _database.AddParameter("@OrderID", OrderID, true);
                _database.ExecuteSP_NO_Return("SiteOrder_DeleteByOrder");
            }

            public SiteOrder SiteOrder_Get(int SiteOrderID)
            {
                _database.ClearParameter();
                _database.AddParameter("@SiteOrderID", SiteOrderID);

                // Get the datatable from the database store in dt
                var dt = _database.ExecuteSP_DataTable("SiteOrder_Get");
                if (dt is object)
                {
                    if (dt.Rows.Count > 0)
                    {
                        var oSiteOrder = new SiteOrder();
                        oSiteOrder.IgnoreExceptionsOnSetMethods = true;
                        oSiteOrder.SetSiteOrderID = dt.Rows[0]["SiteOrderID"];
                        oSiteOrder.SetSiteID = dt.Rows[0]["SiteID"];
                        oSiteOrder.SetOrderID = dt.Rows[0]["OrderID"];
                        oSiteOrder.Exists = true;
                        return oSiteOrder;
                    }
                    else
                    {
                        return null;
                    }
                }
                else
                {
                    return null;
                }
            }

            public SiteOrder SiteOrder_GetForOrder(int OrderID)
            {
                _database.ClearParameter();
                _database.AddParameter("@OrderID", OrderID);

                // Get the datatable from the database store in dt
                var dt = _database.ExecuteSP_DataTable("SiteOrder_GetForOrder");
                if (dt is object)
                {
                    if (dt.Rows.Count > 0)
                    {
                        var oSiteOrder = new SiteOrder();
                        oSiteOrder.IgnoreExceptionsOnSetMethods = true;
                        oSiteOrder.SetSiteOrderID = dt.Rows[0]["SiteOrderID"];
                        oSiteOrder.SetSiteID = dt.Rows[0]["SiteID"];
                        oSiteOrder.SetOrderID = dt.Rows[0]["OrderID"];
                        oSiteOrder.Exists = true;
                        return oSiteOrder;
                    }
                    else
                    {
                        return null;
                    }
                }
                else
                {
                    return null;
                }
            }

            public DataView SiteOrder_GetAll()
            {
                _database.ClearParameter();
                var dt = _database.ExecuteSP_DataTable("SiteOrder_GetAll");
                dt.TableName = Sys.Enums.TableNames.tblSiteOrder.ToString();
                return new DataView(dt);
            }

            public SiteOrder Insert(SiteOrder oSiteOrder, SqlTransaction trans)
            {
                var Command = new SqlCommand();
                Command.CommandText = "SiteOrder_Insert";
                Command.CommandType = CommandType.StoredProcedure;
                Command.Transaction = trans;
                Command.Connection = trans.Connection;
                Command.Parameters.AddWithValue("@OrderID", oSiteOrder.OrderID);
                Command.Parameters.AddWithValue("@SiteID", oSiteOrder.SiteID);
                oSiteOrder.SetSiteOrderID = Sys.Helper.MakeIntegerValid(Command.ExecuteScalar());
                oSiteOrder.Exists = true;
                return oSiteOrder;
            }

            public SiteOrder Insert(SiteOrder oSiteOrder)
            {
                _database.ClearParameter();
                AddSiteOrderParametersToCommand(ref oSiteOrder);
                oSiteOrder.SetSiteOrderID = Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("SiteOrder_Insert"));
                oSiteOrder.Exists = true;
                return oSiteOrder;
            }

            public void Update(SiteOrder oSiteOrder)
            {
                _database.ClearParameter();
                _database.AddParameter("@SiteOrderID", oSiteOrder.SiteOrderID, true);
                AddSiteOrderParametersToCommand(ref oSiteOrder);
                _database.ExecuteSP_NO_Return("SiteOrder_Update");
            }

            private void AddSiteOrderParametersToCommand(ref SiteOrder oSiteOrder)
            {
                {
                    var withBlock = _database;
                    withBlock.AddParameter("@SiteID", oSiteOrder.SiteID, true);
                    withBlock.AddParameter("@OrderID", oSiteOrder.OrderID, true);
                }
            }


            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}